src/test/java/org/jenkinsci/plugins/durabletask/PowershellScriptTest.java

```diff
@@ -0,0 +1,168 @@
+        while (c.exitStatus(ws, launcher) == null) {
+            c.writeLog(ws, tos);
+            Thread.sleep(100);
+        }
+        c.writeLog(ws, tos);
+        assertEquals(Integer.valueOf(1), c.exitStatus(ws, launcher));
+        String log = baos.toString();
+        assertTrue(log, log.contains("Hello, World!"));
+        c.cleanup(ws);
+    }
+    
+    @Test public void implicitExit() throws Exception {
+        Controller c = new PowershellScript("Write-Output \"Success!\";").launch(new EnvVars(), ws, launcher, listener);
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        TeeOutputStream tos = new TeeOutputStream(baos, System.err);
+        while (c.exitStatus(ws, launcher) == null) {
+            c.writeLog(ws, tos);
+            Thread.sleep(100);
+        }
+        c.writeLog(ws, tos);
+        assertEquals(Integer.valueOf(0), c.exitStatus(ws, launcher));
+        String log = baos.toString();
+        assertTrue(log, log.contains("Success!"));
+        c.cleanup(ws);
+    }
+    
+    @Test public void implicitError() throws Exception {
+        Controller c = new PowershellScript("MyBogus-Cmdlet").launch(new EnvVars(), ws, launcher, listener);
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        TeeOutputStream tos = new TeeOutputStream(baos, System.err);
+        while (c.exitStatus(ws, launcher) == null) {
+            c.writeLog(ws, tos);
+            Thread.sleep(100);
+        }
+        c.writeLog(ws, tos);
+        assertTrue(c.exitStatus(ws, launcher).intValue() != 0);
+        c.cleanup(ws);
+    }
+    
+    @Test public void explicitError() throws Exception {
+        DurableTask task = new PowershellScript("Write-Output \"Hello, World!\"; throw \"explicit error\";");
+        task.captureOutput();
+        Controller c = task.launch(new EnvVars(), ws, launcher, listener);
+        while (c.exitStatus(ws, launcher) == null) {
+            Thread.sleep(100);
+        }
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        c.writeLog(ws, baos);
+        assertTrue(c.exitStatus(ws, launcher).intValue() != 0);
+        assertThat(baos.toString(), containsString("explicit error"));
+        c.cleanup(ws);
+    }
+    
+    @Test public void verbose() throws Exception {
+        DurableTask task = new PowershellScript("$VerbosePreference = \"Continue\"; Write-Verbose \"Hello, World!\"");
+        task.captureOutput();
+        Controller c = task.launch(new EnvVars(), ws, launcher, listener);
+        while (c.exitStatus(ws, launcher) == null) {
+            Thread.sleep(100);
+        }
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        c.writeLog(ws, baos);
+        assertEquals(0, c.exitStatus(ws, launcher).intValue());
+        assertThat(new String(c.getOutput(ws, launcher)), containsString("Hello, World!"));
+        c.cleanup(ws);
+    }
+        
+    @Test public void output() throws Exception {
```

Comment:
This test doesn't seem necessary for powershell, since you already test output in other tests. I believe this test was there for batch files because of the ability to turn off echoing of the command via \@echo.